src/openforms/forms/tests/test_data_migrations.py

```diff
@@ -73,3 +74,60 @@ def test_open_when_empty_is_false(self):
             component = components["editgrid4"]
 
             self.assertFalse(component["openWhenEmpty"])
+
+
+class TestFormVariableDataType(TestMigrations):
+    """Assert that if type(form_component) == date, then type(form_varialbe) == date"""
+
+    migrate_from = "0071_merge_20230213_1106"
+    migrate_to = "0072_check_form_variable_datatype"
+    app = "forms"
+
+    def setUpBeforeMigration(self, apps):
+        Form = apps.get_model("forms", "Form")
+        FormDefinition = apps.get_model("forms", "FormDefinition")
+        FormVariable = apps.get_model("forms", "FormVariable")
+
+        form = Form.objects.create(name="form")
+        configuration = {
+            "components": [
+                {"key": "field1", "type": "datetime"},
+                {"key": "field2", "type": "date"},
+            ]
+        }
+        form_def = FormDefinition.objects.create(
+            name="form_def",
+            configuration=configuration,
+        )
+        FormVariable.objects.create(
+            key="field1",
+            form=form,
+            form_definition=form_def,
+            data_type=FormVariableDataTypes.datetime,
+        )
+        FormVariable.objects.create(
+            key="field2",
+            form=form,
+            form_definition=form_def,
+            data_type=FormVariableDataTypes.datetime,
+        )
+        FormVariable.objects.create(
+            key="unrelated",
+            form=form,
+            data_type=FormVariableDataTypes.datetime,
+        )
+
+    def test_form_variable_datatype(self):
+        FormVariable = self.apps.get_model("forms", "FormVariable")
+        form_var1 = FormVariable.objects.filter(key="field1").first()
+        form_var2 = FormVariable.objects.filter(key="field2").first()
+        form_var3 = FormVariable.objects.filter(key="unrelated").first()
```

Comment:
for the future - since there's only one expected match, you can do `FormVariable.objects.get(key="...")` instead.